ddtrace/internal/telemetry/__init__.py

```diff
@@ -105,6 +101,15 @@ def get_config(
     return val
 
 
+if get_config("DD_INSTRUMENTATION_TELEMETRY_ENABLED", True, asbool, report_telemetry=False):
```

Comment:
We should be able to check this from the `config` object in `ddtrace.settings._config` (perhaps some things would have to be moved to avoid a circular import though?)

Reply:
Unfortunately, I do not think checking the `config` object is going to work in this case. The `config` object has a custom `__setatter__` which will call `telemetry_writer.add_configuration` if telemetry is enabled. Meaning, the `config` object relies on the `telemetry_writer` being fully initialized.

I do agree that using `get_config` directly here is adding unneeded overhead. Fortunately at least, `ddtrace.settings._config` actually imports and uses the `get_config` method.

I'm thinking the best way to handle this is actually going to be the reverse. Instead of getting the value from the `config`, let the `config` get the value from here. That way the work isn't being repeated.

I'll go ahead and code that up. Let me know what you think of it.

Reply:
> The config object has a custom __setatter__ which will call telemetry_writer.add_configuration if telemetry is enabled. Meaning, the config object relies on the telemetry_writer being fully initialized.

Ah I see. That's why I'd love to see a different architecture for telemetry where these different parts (configuration, writer, etc...) are not so tangled up together. I guess we'll have to leave that to follow-up work 🙁